core/services/relay/evm/request_round_tracker.go

```diff
@@ -0,0 +1,209 @@
+package evm
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	gethCommon "github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/libocr/gethwrappers2/ocr2aggregator"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+	"github.com/smartcontractkit/sqlx"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/log"
+	offchain_aggregator_wrapper "github.com/smartcontractkit/chainlink/core/internal/gethwrappers2/generated/offchainaggregator"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/ocrcommon"
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// RequestRoundTracker subscribes to new request round logs.
+type RequestRoundTracker struct {
+	utils.StartStopOnce
+
+	ethClient        evmclient.Client
+	contract         *offchain_aggregator_wrapper.OffchainAggregator
+	contractFilterer *ocr2aggregator.OCR2AggregatorFilterer
+	logBroadcaster   log.Broadcaster
+	jobID            int32
+	logger           logger.Logger
+	odb              RequestRoundDB
+	q                pg.Q
+	blockTranslator  ocrcommon.BlockTranslator
+
+	// Start/Stop lifecycle
+	ctx             context.Context
+	ctxCancel       context.CancelFunc
+	unsubscribeLogs func()
+
+	// LatestRoundRequested
+	latestRoundRequested ocr2aggregator.OCR2AggregatorRoundRequested
+	lrrMu                sync.RWMutex
+
+	// LatestBlockHeight
+	latestBlockHeight   int64
+	latestBlockHeightMu sync.RWMutex
+}
+
+// NewRequestRoundTracker makes a new RequestRoundTracker
+func NewRequestRoundTracker(
+	contract *offchain_aggregator_wrapper.OffchainAggregator,
+	contractFilterer *ocr2aggregator.OCR2AggregatorFilterer,
+	ethClient evmclient.Client,
+	logBroadcaster log.Broadcaster,
+	jobID int32,
+	logger logger.Logger,
+	db *sqlx.DB,
+	odb RequestRoundDB,
+	chain ocrcommon.Config,
+) (o *RequestRoundTracker) {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &RequestRoundTracker{
+		utils.StartStopOnce{},
+		ethClient,
+		contract,
+		contractFilterer,
+		logBroadcaster,
+		jobID,
+		logger,
+		odb,
+		pg.NewQ(db, logger, chain),
+		ocrcommon.NewBlockTranslator(chain, ethClient, logger),
+		ctx,
+		cancel,
+		nil,
+		ocr2aggregator.OCR2AggregatorRoundRequested{},
+		sync.RWMutex{},
+		-1,
+		sync.RWMutex{},
```

Comment:
This is pretty large, maybe worth using named fields and dropping the zero values?